lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -390,6 +403,9 @@ def _call(self, method: Callable, json_data: JSON):
         except UnauthorizedAccess as e:
             self.logger.warning("{}", e)
             abort(e.http_status, message=str(e))
+        except InvalidAPIRequest as e:
+            self.logger.warning("{}", e)
+            abort(e.status, message=str(e))
```

Comment:
I think that this catch (and the definition of the exception) should be a layer further out, in `ApiBase._dispatch` (`lib/pbench/server/api/resources/__init__.py`). The various subclasses of that class (including this one) shouldn't need to provide handlers unless they need to clean up locally, which I don't think is the case here.

Reply:
Since there is a generic exception handler written for preprocess stage:

> `        except Exception as e:
>             self.logger.exception("{} preprocess failed: {}", klasname, e)
>             abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")`

Therefore all the exceptions that occurred during `preprocess()`, get handled here itself. So only I updated here instead of `ApiBase._dispatch`. 
But I get the context here, I will update the suggested changes accordingly